pkg/api/v20191231preview/openshiftcluster_validate_test.go

```diff
@@ -0,0 +1,689 @@
+			}
+		})
+	}
+}
+
+func TestWorkerProfileValidate(t *testing.T) {
+	subID := uuid.NewV4().String()
+	subnet2ID := fmt.Sprintf("/subscriptions/%s/resourcegroups/test-vnet/providers/Microsoft.Network/virtualNetworks/other-vnet/subnets/other", subID)
+	mp := MasterProfile{
+		SubnetID: subnetID,
+		VMSize:   VMSizeStandardD8sV3,
+	}
+	tests := []struct {
+		name       string
+		Name       string
+		VMSize     VMSize
+		DiskSizeGB int
+		SubnetID   string
+		Count      int
+		wantErr    bool
+		errMsg     string
+	}{
+		{
+			name:       "pass",
+			Name:       "worker",
+			VMSize:     VMSizeStandardD4sV3,
+			DiskSizeGB: 128,
+			SubnetID:   subnetID,
+			Count:      12,
+		},
+		{
+			name:       "disk too small",
+			Name:       "worker",
+			VMSize:     VMSizeStandardD4sV3,
+			DiskSizeGB: 100,
+			SubnetID:   subnetID,
+			Count:      12,
+			wantErr:    true,
+			errMsg:     "The provided worker disk size",
+		},
+		{
+			name:       "count too small",
+			Name:       "worker",
+			VMSize:     VMSizeStandardD4sV3,
+			DiskSizeGB: 134,
+			SubnetID:   subnetID,
+			Count:      2,
+			wantErr:    true,
+			errMsg:     "The provided worker count '2' is invalid",
+		},
+		{
+			name:       "count too big",
+			Name:       "worker",
+			VMSize:     VMSizeStandardD4sV3,
+			DiskSizeGB: 134,
+			SubnetID:   subnetID,
+			Count:      21,
+			wantErr:    true,
+			errMsg:     "The provided worker count '21' is invalid",
+		},
+		{
+			name:       "subnet different to master",
+			Name:       "worker",
+			VMSize:     VMSizeStandardD4sV3,
+			DiskSizeGB: 134,
+			SubnetID:   subnet2ID,
+			Count:      20,
+			wantErr:    true,
+			errMsg:     "must be in the same vnet as master VM subnet",
+		},
+		{
+			name:       "must be worker",
+			Name:       "buzzy-bee",
+			VMSize:     VMSizeStandardD4sV3,
+			DiskSizeGB: 134,
+			SubnetID:   subnetID,
+			Count:      20,
+			wantErr:    true,
+			errMsg:     "The provided worker name 'buzzy-bee' is invalid",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			wp := &WorkerProfile{
+				Name:       tt.Name,
+				VMSize:     tt.VMSize,
+				DiskSizeGB: tt.DiskSizeGB,
+				SubnetID:   tt.SubnetID,
+				Count:      tt.Count,
+			}
+			err := wp.validate(".path", &mp)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("WorkerProfile.validate() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if tt.wantErr && err != nil && !strings.Contains(err.Error(), tt.errMsg) {
+				t.Errorf("error must contain %s, actual error %v", tt.errMsg, err)
+			}
+
+		})
+	}
+}
+
+func TestOpenShiftClusterValidateDelta(t *testing.T) {
+	old := &OpenShiftCluster{
+		Name:       clusterName,
+		ID:         resID,
+		Type:       "Microsoft.RedHatOpenShift/openShiftClusters",
+		Location:   "eastus",
+		Tags:       Tags{"foo": "fee"},
+		Properties: goodProps,
+	}
+
+	tests := []struct {
+		name    string
+		f       func(change *OpenShiftCluster)
+		wantErr bool
```

Comment:
Can you standardise throughout on testing the specific error returned? i.e. no strings.Contains or wantErr bool?

Reply:
sure